lib/lib.go

```diff
@@ -790,6 +790,31 @@ func (inst *Instance) Connect(ctx context.Context) (err error) {
 	return nil
 }
 
+// Access returns the AccessMethods that Instance has registered
+func (inst *Instance) Access() AccessMethods {
+	return AccessMethods{d: inst}
+}
+
+// Config returns the ConfigMethods that Instance has registered
+func (inst *Instance) Config() *ConfigMethods {
+	return &ConfigMethods{d: inst}
+}
+
+// Dataset returns the DatasetMethods that Instance has registered
+func (inst *Instance) Dataset() *DatasetMethods {
+	return &DatasetMethods{inst: inst}
+}
+
+// Filesys returns the FSIMethods that Instance has registered
+func (inst *Instance) Filesys() *FSIMethods {
+	return &FSIMethods{d: inst}
+}
+
+// Transform returns the TransformMethods that Instance has registered
+func (inst *Instance) Transform() *TransformMethods {
+	return &TransformMethods{d: inst}
+}
```

Comment:
do these need to return pointers? I'd prefer they not

Reply:
It was done to be consistent with the old Method style structs, but you're right that there's no need for that.